php-practice/section3.php

```diff
@@ -183,7 +191,11 @@ function reverseArray($originalArray) { // Notice that it takes an array as an a
 
 // Alright, let's see this baby in action! Uncomment the print_r() line and run this script.
 $reversedNames = reverseArray($names);
-// print_r($reversedNames);
+print_r($reversedNames);
 
 // In your own words, explain how the function I created works:
-// [YOUR ANSWER HERE]
+//
+// first we create our array $names. Then we want to run a function that passes the argument $originalArray. Inside our curly braces we put a while loop
+// since we wnat the code do run "while" a condition we specify remains true. In this case, we specify that "while" our argument ($originalArray)
+// evaluates to "not empty" or "false" then we want the loop to "pop" a value off the end of the array and put in the new one, specified by the array $newArray[]
+// We then want to "return" the value in our $newArray and repeat the loop until the $originalArray is "empty" fulfilling our "while" loop.
```

Comment:
Great job explaining this -- you're mostly understanding the flow of the code, which is important.

### To answer your second question 

(Understanding this line: `$reversedNames = reverseArray($names);`)

One correction: you said...

> We then want to "return" the value in our $newArray and repeat the loop...

But notice that the `return` only happens `after` the loop is finished.

Here's the code from inside that function:

```PHP
while (!empty($originalArray)) {
  $newArray[] = array_pop($originalArray);
}

return $newArray;
```

Remember, the code that gets looped over is the stuff inside the curly brackets. That code keeps getting ran until the array is empty, and it's only *then* that the function returns a value.

Note about `return`: Whenever a function `return`s something, the function immediately stops executing and "returns" whatever you told it to return back wherever the function was called.

So your question was about what this line does:

```PHP
$reversedNames = reverseArray($names);
```

Well, the expression `reverseArray($names)` gets replaced by whatever the `return` value of that function is. Or to say it another way, it *evaluates* to that return value. (In other words, the PHP interpreter runs that function, and then drops the return value in place of `reverseArray($names)`.)

So whatever `$newArray` is when we return it -- that's the value that `$reversedNames` is set to whenever we run

```PHP
$reversedNames = reverseArray($names);
```